CMake/Version.cmake

```diff
@@ -0,0 +1,72 @@
+
+# Try to figure out  versioning scheme for dynamic libraries
+# based on the system we're compiling for
+function(determine_version_type _out_version_type_name)
+
+        set(_ver_type "none")
+
+        if(WIN32 OR CYGWIN)
+                set(_ver_type "windows")
+        else()
+                # this part tries to mimic autotools as close as possible
+                if(CMAKE_SYSTEM_NAME STREQUAL "AIX" OR
+                   CMAKE_SYSTEM_NAME STREQUAL "Linux" OR
+                   CMAKE_SYSTEM_NAME STREQUAL "GNU/kFreeBSD" OR
+                   CMAKE_SYSTEM_NAME STREQUAL "Haiku" OR
+                   CMAKE_SYSTEM_NAME STREQUAL "SunOS")
+                        set(_ver_type "linux")
+                elseif(CMAKE_SYSTEM_NAME STREQUAL "HP-UX" OR
+                       CMAKE_SYSTEM_NAME STREQUAL "NetBSD" OR
+                       CMAKE_SYSTEM_NAME STREQUAL "OpenBSD")
+                        set(_ver_type "sunos")
```

Comment:
while a few other OSes are type "sunos" ? That's... weird.

Reply:
> while a few other OSes are type "sunos" ? That's... weird.

@jzakrzewski can you address this, based on my read it looks like you have _ver_type as only used to make the versioning format, so for example hp-ux uses sunos versioning.. is that right?